File: python/82_armstrong_numbers.py
from math import pow
from sys import argv

with open(argv[1], 'r') as fh:
    for line in fh.readlines():
        line = line.strip()
        if line == '':
            continue

        num = int(line)
        length = len(line)
        sum = 0
        for c in line:
            sum += int(pow(int(c), length))

        print num == sum

File: perl/82_armstrong_numbers.pl
open my $fh, '<', $ARGV[0];

for my $line ( <$fh> ) {
	$line =~ s/[\n\r]+//;
	next if $line eq '';

	my $sum = 0;
	for my $c ( split //, $line ) {
		$sum += $c ** length $line;
	}

	print $sum == $line ? "True\n" : "False\n";
}

close $fh;

exit 0;

